upup/pkg/fi/cloudup/dotasks/volume.go

```diff
@@ -108,12 +110,26 @@ func (_ *Volume) RenderDO(t *do.DOAPITarget, a, e, changes *Volume) error {
 		return nil
 	}
 
+	tagArray := []string{}
+
+	klog.V(2).Info("Looping DO tag arrays")
+	for k, v := range e.Tags {
+		s := []string{k, v}
+
+		// DO tags don't accept =. Separate the key and value with an ":"
+		strJoin := strings.Join(s, ":")
+		klog.V(2).Infof("DO - Join the volume tag - %s", strJoin)
+		tagArray = append(tagArray, strJoin)
```

Comment:
We could also do a direct

```go
tagArray = append(tagArray, fmt.Sprintf("%s:%s", k, v)
```

here and drop `s` and `strJoin` since we're always joining a pair only.

Reply:
Sounds great, updated !